OrbitQt/ConnectionConfiguration.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_QT_CONNECTION_CONFIGURATION_H_
+#define ORBIT_QT_CONNECTION_CONFIGURATION_H_
+
+#include <memory>
+#include <utility>
+
+#include "DeploymentConfigurations.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitClientData/ProcessData.h"
+#include "OrbitClientServices/ProcessManager.h"
+#include "OrbitGgp/Instance.h"
+#include "OrbitSsh/Context.h"
+#include "OrbitSsh/Credentials.h"
+#include "grpcpp/channel.h"
+#include "servicedeploymanager.h"
+
+namespace orbit_qt {
+
+struct StadiaConnection {
+ public:
+  explicit StadiaConnection(const OrbitSsh::Context* ssh_context,
+                            const ServiceDeployManager::GrpcPort& grpc_port,
+                            const DeploymentConfiguration* deployment_configuration)
+      : ssh_context(ssh_context),
+        grpc_port(grpc_port),
+        deployment_configuration(deployment_configuration) {
+    CHECK(ssh_context != nullptr);
+    CHECK(deployment_configuration != nullptr);
+  }
+  ~StadiaConnection() {
+    if (process_manager != nullptr) process_manager->Shutdown();
+  }
```

Comment:
When defining a destructor you should also define copy/move-constructors/assignment operators. (Rule of 5!)

Reply:
`Shutdown` should probably called by ProcessManager's destructor. Forcing that requirement on the user doesn't feel right.

Reply:
(BTW, maybe you didn't notice but by defining this destructor this struct gets non-movable and it's not copyable because of its members which means you have to head-allocate it all the time what you probably don't want.)

Reply:
See #1547